### ISSUE TITLE
[PDKI-712] Data Mask AI - Coding Phase - EP -  Scene Text Editing research

### ISSUE THREAD

**Issue by jinijohn26** [5Plus-Test/jiramigrationtest#387]
> Migrated from Jira: [PDKI-712](https://pentadoc.atlassian.net/browse/PDKI-712)

Scene Text Editing isthe task expression/name for text editing with automatic style approximation

[UCSB-NLP-Chang/DiffSTE|https://github.com/UCSB-NLP-Chang/DiffSTE]

[youdao-ai/SRNet: A tensorflow reproducing of paper “Editing Text in the wild”|https://github.com/youdao-ai/SRNet]

[prasunroy/stefann: :fire: [CVPR 2020] STEFANN: Scene Text Editor using Font Adaptive Neural Network (official code).|https://github.com/prasunroy/stefann?tab=readme-ov-file]

HQ: I-0063: Produktentwicklung Data Mask AI

---
### Checklist

- [x] test out existing STE models
- [x] research other models
- [x] research fine-tuning methods
- [x] share results with @712020:60b8d4bb-6f66-40fc-8a87-7cb06132593f

---
### Comments

**Philipp Horn** (07/Mar/25 4:44 PM):
> Viony and I had a short meeting with Boris to give us access to the Google Cloud GPU resources. They could be required for testing the STE models.

**Philipp Horn** (07/Mar/25 5:17 PM):
> I tested DiffSTE for a bit.
It needs some work to be able to run it, since Runway removed their Stable Diffusion model from Hugging Face. But, with a few changes I was able to reproduce their samples.

**Viony Tengguna** (13/Mar/25 9:30 AM):
> TextCTRL  
[weichaozeng/TextCtrl: [2024-NeurIPS] TextCtrl: Diffusion-based Scene Text Editing with Prior Guidance Control|https://github.com/weichaozeng/TextCtrl?tab=readme-ov-file]

* The code is working locally and managed to change sample images texts to similar/almost similar font as the original characters
* Run on float/bfloat16
* Will try more on our documents

Got some others STEs to try out based on discussion result with Boris:

[tyxsspa/AnyText: Official implementation code of the paper <AnyText: Multilingual Visual Text Generation And Editing>|https://github.com/tyxsspa/AnyText]

[unilm/textdiffuser at master · microsoft/unilm|https://github.com/microsoft/unilm/tree/master/textdiffuser]

**Viony Tengguna** (13/Mar/25 4:07 PM):
> SRNet checkpoint link - not available, didn’t find any replacements for it

AnyText - huggingface not available anymore. Found latest repo and I’m currently trying this out [tyxsspa/AnyText2: Official implementation code of the paper <AnyText2: Visual Text Generation and Editing With Customizable Attributes>|https://github.com/tyxsspa/AnyText2]

**Viony Tengguna** (14/Mar/25 9:49 AM):
> Anytext2 environment setup problem solved (they use xformers, pytorch, and tensorflow at once). 

Cannot rely on ymal environment since it will cause the crash. Therefore the installation needs to be done one by one to solve the error crash.

As of now, I’m able to run it untiil the GUI comes up but got some error when it runs. Judging by the environment library suggested versions, the code is actually a bit older but the library got some update recently. Therefore some functions are unable to be used anymore. Either I should try to look for the library with troubles and downgrade the versions or change the functions :'

**Philipp Horn** (17/Mar/25 5:50 PM):
> We are tracking the research status on the following Miro board:
[https://miro.com/app/board/uXjVIQ3nDPM=/?moveToWidget=3458764621017721878&cot=14|https://miro.com/app/board/uXjVIQ3nDPM=/?moveToWidget=3458764621017721878&cot=14|smart-link]

**Philipp Horn** (25/Mar/25 3:09 PM):
> After another round of testing we found that TextCTRL is the best tool for the job!

* In the Google Cloud VM provided by Boris, TextCTRL only requires around 6.6GB of GPU memory. Much better than the 14GB it needed on Viony’s laptop.
* DiffSTE is not able to match the style of the text it is replacing, but rather tries to match it to the surrounding text style.
* DiffSTE has many features that we do not need (like creating a style from natural language input) making it unnecessarily complex.

The only thing TextCTRL (and all other tools we tested) is not able to do, is insert text that contain an umlaut or ß. A work around for this problem can be to restrict the replacement library with words that do not contain an umlaut or ß. Alternatively, one can retrain the model with an extended dataset that contains words with umlaut and ß.
